src/Bicep.Core/Emit/TemplateWriter.cs

```diff
@@ -549,6 +584,87 @@ private static bool ShouldGenerateDependsOn(ResourceDependency dependency)
             }
         }
 
+        private void EmitSymbolicNameDependsOnEntry(JsonTextWriter jsonWriter, ExpressionEmitter emitter, SyntaxBase newContext, ResourceDependency dependency)
+        {
+            switch (dependency.Resource)
+            {
+                case ResourceSymbol resourceDependency:
+                    var resource = context.SemanticModel.ResourceMetadata.TryLookup(resourceDependency.DeclaringSyntax) ??
+                        throw new ArgumentException($"Unable to find resource metadata for dependency '{dependency.Resource.Name}'");
+
+                    switch ((resourceDependency.IsCollection, dependency.IndexExpression))
+                    {
+                        case (false, _):
+                            jsonWriter.WriteValue(resourceDependency.Name);
```

Comment:
Should we add `Debug.Assert(dependency.IndexExpression is null)` here and below to catch bugs?

Reply:
Done